### PR TITLE
Making Android build simpler.

### PR DESCRIPTION
**Pull request by mrbobbybobberson**
* Fixing up app permissions automatically when PlatformSwitcher is used to select the platform.
  * Automatically embedding an AndroidManifest.xml with the correct application activity.
  * Updating documentation steps to leave out the "Export" instruction, leaving it up to the user whether they build fully in Unity or with Unity + Android Studio.
  * Also added an automatic setting for iOS to require ARKit-capable OS versions.

(cherry picked from commit 8e70cb7de88b36fba1f6845200f196528f8c25b9)

### CODE REVIEW

**Comment by mrbobbybobberson** [comment]
Hey @chrisfromwork, @andreiborodin, @matthejo, I was feeling slowed down by always having to remember to export the android project and then manually edit the manifest in Android Studio before I built, so I went down a rabbit hole trying to make it possible to fully build in Unity, and this is what I came up with.  Is this something y'all would like to take?  Or should I keep it my private bag of tricks?

File: src/SpectatorView.Unity/Assets/SpectatorView/Scripts/ScreenRecording/Plugins/Android/AndroidManifestTemplate.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="UNITY WILL REPLACE" xmlns:tools="http://schemas.android.com/tools" android:installLocation="preferExternal">
+  <supports-screens android:smallScreens="true" android:normalScreens="true" android:largeScreens="true" android:xlargeScreens="true" android:anyDensity="true" />
+  <application android:theme="@style/UnityThemeSelector" android:icon="@mipmap/app_icon" android:label="@string/app_name" android:isGame="true" android:banner="@drawable/app_banner">
+    <activity android:label="@string/app_name" android:screenOrientation="portrait" android:launchMode="singleTask" android:configChanges="mcc|mnc|locale|touchscreen|keyboard|keyboardHidden|navigation|orientation|screenLayout|uiMode|screenSize|smallestScreenSize|fontScale|layoutDirection|density" android:hardwareAccelerated="false" android:name="Microsoft.MixedReality.SpectatorView.Unity.ScreenRecorderActivity">
```
Comment (chrisfromwork) [inline src/SpectatorView.Unity/Assets/SpectatorView/Scripts/ScreenRecording/Plugins/Android/AndroidManifestTemplate.xml]: Can user permissions be declared in this template? We were declaring WRITE_EXTERNAL_STORAGE and RECORD_AUDIO in our previously existing manifest.

File: doc/SpectatorView.Setup.md [hunk 2]
```diff
@@ -141,21 +141,12 @@ If you are building Azure Spatial Anchors on iOS, you will need to take some add
...
-8. Update the AndroidManifest.xml in Android Studio to use the `Microsoft.MixedReality.SpectatorView.Unity.ScreenRecorderActivity` class compared to the UnityPlayerActivity as the application activity.
-
-![Marker](images/AndroidManifestActivity.png)
-
-9. Update the AndroidManifest.xml in Android Studio to contain `android.permission.CAMERA`, `android.permission.INTERNET`, `android.permission.RECORD_AUDIO` and `android.permission.WRITE_EXTERNAL_STORAGE` uses-permissions.
-
-![Marker](images/AndroidManifestPermission.png)
-
-10. Build and deploy the application through Android Studio to your desired device. If it is a new Android development device, you may need to enable developer options and debugging. For information on setting up your Android device for debugging, see [here](https://developer.android.com/studio/debug/dev-options).
+6. Build and deploy the application. If you're targeting a new Android development device, you may need to enable developer options and debugging. For information on setting up your Android device for debugging, see [here](https://developer.android.com/studio/debug/dev-options).
```
Comment (chrisfromwork) [inline doc/SpectatorView.Setup.md]: I'm all for simplifying this process. We may want to keep the previous documentation in a note here or a troubleshooting step if folks already have an existing android manifest.

File: src/SpectatorView.Unity/Assets/SpectatorView.Editor/Scripts/PlatformSwitcherEditor.cs [hunk 3]
```diff
@@ -40,20 +45,84 @@ public override void OnInspectorGUI()
...
+                    Directory.CreateDirectory(destManifestDirectoryPath);
+                    File.Copy(sourceManifestPath, destManifestPath);
+                }
+
+                var manifest = XElement.Load(destManifestPath);
+                var activities = manifest.XPathSelectElements("application/activity").ToArray();
+
+                if (activities.Length != 1)
+                {
+                    throw new System.Exception($"Expected 1 application activity, but got {activities.Length}.");
```
Comment (chrisfromwork) [inline src/SpectatorView.Unity/Assets/SpectatorView.Editor/Scripts/PlatformSwitcherEditor.cs]: maybe log error compared to throw if this is ran in the editor?

File: src/SpectatorView.Unity/Assets/SpectatorView/Scripts/ScreenRecording/Plugins/Android/AndroidManifestTemplate.xml (hunk 1 above)
Comment (mrbobbybobberson) [reply src/SpectatorView.Unity/Assets/SpectatorView/Scripts/ScreenRecording/Plugins/Android/AndroidManifestTemplate.xml]: Both of those will be automatically added by Unity to the final manifest.  I debated putting comments in this template to explain, but then those comments end up in the final manifest, too, so I opted to leave them out.  I'm happy to put comments back into the manifest though if you'd prefer.

File: doc/SpectatorView.Setup.md (hunk 2 above)
Comment (mrbobbybobberson) [reply doc/SpectatorView.Setup.md]: I'll add a note for people who have their own manifests.

File: src/SpectatorView.Unity/Assets/SpectatorView.Editor/Scripts/PlatformSwitcherEditor.cs (hunk 3 above)
Comment (mrbobbybobberson) [reply src/SpectatorView.Unity/Assets/SpectatorView.Editor/Scripts/PlatformSwitcherEditor.cs]: I need to try/catch to make sure the potential disk operation exceptions get caught and logged with appropriate context, so I figured I'd throw for all errors and catch all errors (including these) and apply the same context to the logging message in the catch block.

**Pull request merged**